accessibility_api/accessibility_lib/utils/WinUtil.py

```diff
@@ -0,0 +1,217 @@
+                                                 simple_elements)
+                else:
+                    acc_objs.append(None)
+            return acc_objs
+        else:
+            raise ValueError('Failed to get accessible children')
+
+    @staticmethod
+    def _wrap_simple_element(acc_ptr, childid, simple_elements):
+        """
+        Associate simple element and parent accessible object
+        """
+
+        if acc_ptr not in simple_elements:
+            simple_elements[acc_ptr] = [childid]
+        else:
+            simple_elements[acc_ptr].append(childid)
+
+    @staticmethod
+    def _traverse(node, visited, search_criteria, simple_elements):
+        """
+        Traverse through accessible tree looking for node with the given ID
+        """
+
+        global TARGET
+
+        if WinUtil.match_criteria(node, search_criteria):
+            TARGET = node
+            TARGET.isSimpleElement = False
+            return
+
+        print_accessible(node)
+
+        # Retrieve simple children or accessible children from node
+        acc_children = WinUtil.accessible_children(node, simple_elements)
+
+        # Traverse through simple elements of node
+        if node in simple_elements:
+            for childid in simple_elements[node]:
+                print_simple(node, childid)
+
+                if WinUtil.match_criteria(node, search_criteria, childid):
+                    TARGET = node
+                    TARGET.isSimpleElement = True
+                    TARGET.childId = childid
+                    return
+
+        # Traverse through accessible objects of node
+        for child in acc_children:
+            if child not in visited:
+                visited.add(node)
+                WinUtil._traverse(child, visited, search_criteria,
+                                  simple_elements)
+
+    @staticmethod
+    def get_indentifiers(params):
+        """
+        Get accessibility indentifiers
+        """
+
+        identifiers = {
+            'Name': params.get('name'),
+            'Role': params.get('role')
+        }
+        filtered_identifiers = {k: v for k, v in identifiers.items() if v}
+
+        return filtered_identifiers
+
+    @staticmethod
+    def match_criteria(node, params, child_id=CHILDID_SELF):
+        """
+        Find matching accessible with given criteria
+        """
+
+        search_criteria = WinUtil.get_indentifiers(params)
+        for criteria in search_criteria:
+            prefix = 'acc'
+            prop_value = getattr(node, prefix + criteria)(child_id)
+            search_value = search_criteria[criteria]
+
+            # If value is a number convert from unicode to int
+            if isinstance(prop_value, int):
+                search_value = int(search_value)
+
+            if prop_value != search_value:
+                return False
+
+        return True
+
+    @staticmethod
+    def _get_test_window(app_window, simple_elements):
```

Comment:
afaict app_window is the name, so lets rename this into test_window_name or something like that.